internal/commands/app/init.go

```diff
@@ -50,16 +55,77 @@ func (cmd *CommandInit) Handler(profile *cli.Profile, ui terminal.UI, clients cl
 }
 
 func (cmd *CommandInit) writeAppFromScratch(wd string) error {
-	/*
-		TODO(REALMC-7886): initialize also the following:
-			- auth/custom_user_data.json: { "enabled": false }
-			- auth/providers.json: {}
-			- data_sources/
-			- http_endpoints/
-			- sync/config.json: { "development_mode_enabled": false }
+	switch cmd.inputs.ConfigVersion {
+	case
+		realm.AppConfigVersion20200603,
+		realm.AppConfigVersion20180301:
+		configAPIKey := map[string]interface{}{
+			"name":     "api-key",
+			"type":     "api-key",
+			"disabled": true,
+		}
+		data, err := local.MarshalJSON(configAPIKey)
+		if err != nil {
+			return err
+		}
+		err = local.WriteFile(filepath.Join(wd, "auth_providers", "api-key.json"), 0666, bytes.NewReader(data))
+		if err != nil {
+			return err
+		}
+		if err := os.MkdirAll(filepath.Join(wd, "functions"), os.ModePerm); err != nil {
+			return err
+		}
+	default:
+		err := local.WriteFile(filepath.Join(wd, "auth", "providers.json"), 0666, bytes.NewReader([]byte("{}")))
```

Comment:
I'd say this looks like roughly the right idea (didn't look too closely at each file being written), but I wonder if it makes sense to try and push this "writing" logic to `(local.App).Write() error`?  I think no matter what this is going to be verbose, but it seems nice to tuck it away behind the `local.App` struct, i'm hoping it makes it a bit clearer in this spot of what the app structure "looks" like per config version